src/jit/emitxarch.cpp

```diff
@@ -3949,6 +3950,33 @@ void emitter::emitIns_R_R_R_I(
     emitCurIGsize += sz;
 }
 
+void emitter::emitIns_R_R_R_R(
+    instruction ins, emitAttr attr, regNumber targetReg, regNumber reg1, regNumber reg2, regNumber reg3)
+{
+    assert(ins == INS_vblendvps || ins == INS_vblendvpd || ins == INS_vpblendvb);
+    assert(UseVEXEncoding());
+    // Currently vex prefix only use three bytes mode.
+    // size = vex + opcode + ModR/M + 1-byte-cns(Reg) = 3 + 1 + 1 + 1 = 6
+    // TODO-XArch-CQ: We should create function which can calculate all kinds of AVX instructions size in future
+    UNATIVE_OFFSET sz = 6;
+
+    // AVX/AVX2 supports 4-reg format for vblendvps/vblendvpd/vpblendvb,
+    // which encodes the fourth register into imm8[7:4]
+    int ival = (reg3 - XMMBASE) << 4; // convert reg3 to ival
```

Comment:
Should we compute the ival later, when we actually emit the instruction bytes?

Reply:
I think computing the `ival` here is good to reuse underlying code.

Reply:
It seems to differ from most of the other emit logic I can find. `emitIns` looks to generally build the id table and `emitOutput` looks to generally transform that into the required format for a given instruction.